### PR TITLE
[cmake] Support INTERFACE_INCLUDE_DIRECTORIES for target dependencies in ROOT_LINKER_LIBRARY

### PR DESCRIPTION
**Pull request by oshadura**
cc: @lmoneta @manolismih 

This PR should fix as well similiar problem in PR https://github.com/root-project/root/pull/7002 (@xvallspl )

### CODE REVIEW

**Comment by oshadura** [comment]
@Axel-Naumann can you take a look whnever you will have time? thanks!

File: roofit/batchcompute/CMakeLists.txt [hunk 1]
```diff
@@ -63,3 +52,4 @@ if (ROOT_PLATFORM MATCHES "linux|macosx" AND CMAKE_SYSTEM_PROCESSOR MATCHES x86_
 
 endif()
 
+ROOT_INSTALL_HEADERS()
```
Comment (hageboeck) [inline roofit/batchcompute/CMakeLists.txt:55]: Why is this needed?

Comment (oshadura) [reply roofit/batchcompute/CMakeLists.txt:55]: I saw that RooBatchCompute.h is used directly in RooFitCore/RoofitMore and since ROOT_LINKER_LIBRARY function is not installing headers, but only .so file I thought you may need it?

Comment (hageboeck) [reply roofit/batchcompute/CMakeLists.txt:55]: Ah, I thought you have to list headers that should be installed. So this just installs all headers from `inc/`? That should be correct.

Comment (oshadura) [reply roofit/batchcompute/CMakeLists.txt:55]: yup, everything from `inc` :+1:

**Review by Axel-Naumann — APPROVED** [review]
